src/figuresClasses.ts

```diff
@@ -1,19 +1,88 @@
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+function round(num: number): number {
+  return Math.floor(num * 100) / 100;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Triangle side should be greater than zero');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('One side shouldn`t be greater than sum of other two');
+    }
+  }
 
+  getArea(): number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
```

Comment:
U can destructure the object to not repeat 'this' such many times)

Reply:
agree